common/sphinx.h

```diff
@@ -133,6 +128,36 @@ struct route_step *process_onionpacket(
 	const size_t assocdatalen
 	);
 
+/**
+ * serialize_per_hop_data - Serialize the per_hop data (for realm 0)
+ * 
+ * @channel_id: the short channel ID
+ * @amt_forward: the amount to forward
+ * @outgoing_cltv: the CLTV value of the outgoing HTLC
+ * @per_hop_data: the serialized data buffer
+ */
+void serialize_per_hop_data(
+	const struct short_channel_id *channel_id,
+	u64 amt_forward,
+	u32 outgoing_cltv,
+	u8 *per_hop_data
+	);
+
+/**
+ * deserialize_per_hop_data - Deserialize the per_hop data (for realm 0)
+ * 
+ * @per_hop_data: the serialized data buffer
+ * @channel_id: the short channel ID
+ * @amt_forward: the amount to forward
+ * @outgoing_cltv: the CLTV value of the outgoing HTLC
+ */
+void deserialize_per_hop_data(
+	const u8 *per_hop_data,
+	struct short_channel_id *channel_id,
+	u64 *amt_forward,
+	u32 *outgoing_cltv
+	);
+
```

Comment:
Here and in other places the style should be a little more like:
```C
void deserialize_per_hop_data(const u8 * per_hop_data,
                              struct short_channel_id * channel_id,
                              u64 * amt_forward,
                              u32 * outgoing_cltv);
```

Reply:
I wonder if we should just have a clang-format linting step in travis...

Reply:
I'm running clang-format on patches that I create, but it's slow progress and having travis check it is really painful if the code mostly does its own thing.

Reply:
I was about to create a new commit to fix this, but then I saw the surrounding, old code was inconsistent w.r.t. this coding style. For instance, the process_onionpacket declaration does not conform to your style suggestion.

Instead of only fixing my new code, it might be better to make one commit that fixes all code. Including such a commit in this pull request would mess up the readability of this pull request, since you'd have a mix of functional and non-functional changes, obscuring the view on where the functional changes are.

So I agree it should be done, but I don't think it should be done right here right now. Let's make a separate pull request that makes the coding style consistent across the entire code base.

Reply:
Shall we maybe commit a `.clang-format` config to format all future code?

Reply:
We could just copy what the linux kernel does:

https://www.kernel.org/doc/html/v4.17/process/clang-format.html

And here's their config file:

https://github.com/torvalds/linux/blob/master/.clang-format

I currently just use `indent -kr -i8 -l80` and then just touch-up the result by hand.